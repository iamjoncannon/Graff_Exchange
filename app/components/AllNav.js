import React from 'react';
import DataNav from './DataNav'
import { Link } from 'react-router-dom'

export default class AllNav extends React.Component {
 
  constructor(props) {
    super(props);

  }

  render() {

    const { pathname } = this.props.location

    return (

      <div className="all-nav">
        
        <Link to="/all/watchlist">

            <span className={pathname.includes("watchlist") ? "selected" : undefined}>
              Portfolio
            </span>   
        
        </Link>
        
        <Link to="/all/transactions">

          <span className={pathname.includes("transactions") ? "selected" : undefined}>
            
              Transactions
          </span>   
  
        </Link>
      </div>
    );
  }
}