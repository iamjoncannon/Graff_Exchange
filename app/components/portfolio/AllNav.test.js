import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import AllNav from './AllNav'
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"
const mockStore = configureStore([thunk]);
import { BrowserRouter as Router } from "react-router-dom";

/*

note- tests will be similar to IndivSelector 

*/

describe("AllNav", ()=>{

    it('1) renders two React Router Link components with span children', ()=>{

        
        
    })

    it("2) if pathname is watchlist, className of Portfolio span is 'selected'", ()=>{

        
        
    })

    it("2) if pathname is transactions, className of Transactions span is 'selected'", ()=>{

        
        
    })
    
})