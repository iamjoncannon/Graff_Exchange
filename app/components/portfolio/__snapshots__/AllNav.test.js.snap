// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllNav 1) renders successfully with data from redux store 1`] = `
<div
  className="all-nav"
>
  <a
    href="/all/watchlist"
    onClick={[Function]}
  >
    <span>
      Portfolio
    </span>
  </a>
  <a
    href="/all/transactions"
    onClick={[Function]}
  >
    <span>
      Transactions
    </span>
  </a>
</div>
`;
